oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/DefaultSegmentWriter.java

```diff
@@ -651,6 +657,17 @@ private RecordId internalWriteStream(@NotNull InputStream stream) throws IOExcep
                         new ByteArrayInputStream(data, 0, n), stream));
                 return writeBlobId(blobId);
             }
+            
+            // handle case in which blob store is not configured and
+            // binariesInlineThreshold < Segment.MEDIUM_LIMIT
+            // store the binaries as small or medium sized value records 
+            
+            data = Arrays.copyOf(data, Segment.MEDIUM_LIMIT);
+            n += read(stream, data, n, Segment.MEDIUM_LIMIT - n);
+            
+            if (n < Segment.MEDIUM_LIMIT) {
+                return writeValueRecord(n, data);
```

Comment:
AFAICS, this line is not covered by a unit test.

Reply:
I added `DefaultSegmentWriterTest#testValueRecord` to specifically test this part.